scripts/kdb.fish

```diff
@@ -0,0 +1,492 @@
+    __fish_kdb_subcommand_includes mount
+    and test (__fish_kdb__number_arguments_input_left) -eq 4
+end
+
+function __fish_kdb_subcommand_needs_storage_plugin -d 'Check if the current subcommand need a storage plugin completion'
+    set -l subcommands editor export import
+
+    not __fish_kdb_subcommand_includes $subcommands
+    and return 1
+
+    __fish_kdb_needs_namespace $subcommands 1
+    and return 1
+    not __input_includes (__fish_kdb_print_storage_plugins)
+end
+
+function __fish_kdb_subcommand_remount_needs_namespace -d 'Check if the subcommand remount needs a namespace completion'
+    not __fish_kdb_subcommand_includes remount
+    and return 1
+
+    set -l number_arguments (__fish_kdb__number_arguments_input_left)
+    test $number_arguments -eq 3 -o $number_arguments -eq 4
+end
+
+function __fish_kdb_subcommand_smount_needs_storage_plugin -d 'Check if the subcommand spec-mount needs a storage plugin completion'
+    __fish_kdb_subcommand_includes smount spec-mount
+    and test (__fish_kdb__number_arguments_input_left) -eq 3
+end
+
+# ============
+# = Printers =
+# ============
+
+function __fish_kdb_print_clause_names -d 'Print a list of possible clause names'
+    printf '%s\n' licence metadata needs placements provides recommends status version
+end
+
+function __fish_kdb_print_metanames -d 'Print a list of possible meta-names'
+    set -l metanames 'order' 'comment' 'line' 'fallback/#' 'override/#' 'namespace/#' 'default' 'context' 'callback/_' 'binary' 'array'
+    set metanames $metanames 'mountpoint' 'infos' 'config' 'opt' 'opt/long' 'env' 'see/#' 'rationale' 'description' 'example'
+    set metanames $metanames 'rename/toupper' 'rename/tolower' 'rename/cut' 'rename/to' 'origname' 'conflict/_' 'array/range' 'required'
+    set metanames $metanames 'logs/_/_' 'warnings' 'error' 'struct' 'check/type' 'check/type/min' 'check/type/max' 'check/format'
+    set metanames $metanames 'check/path' 'check/validation' 'check/validation/message' 'check/validation/match'
+    set metanames $metanames 'check/validation/ignorecase' 'check/validation/invert' 'check/range' 'check/enum' 'check/calculate'
+    set metanames $metanames 'check/condition' 'deprecated' 'internal/<plugin>/*' 'source' 'dependency/control' 'dependency/value'
+    set metanames $metanames 'application/name' 'application/version' 'restrict/write' 'restrict/null' 'restrict/binary' 'restrict/remove'
+    set metanames $metanames 'evaluate/<language>' 'uid' 'gid' 'mode' 'atime' 'mtime' 'ctime' 'spec' 'proc' 'dir' 'user' 'system'
+    set metanames $metanames 'comment/#' 'comment/#/start' 'comment/#/space' 'csv/order' 'crypto/encrypt' 'crypto/salt'
+    printf '%s\n' $metanames
+end
+
+function __fish_kdb_print_namespaces -d 'Print a list of possible namespace completions'
+    set -l namespace (commandline -ct)
+    kdb complete --max-depth=1 -- "$namespace" | string match -vr '(dir|proc|spec|user)$'
```

Comment:
All namespaces do always exist. In the system namespace you are very likely to always find some keys because the built-in version keys do currently live there.

But why is such an assumption valuable?